### ISSUE TITLE
[Bug][zos_job_submit] Exception thrown when submitting a job in LOCAL mode as particular CI user

### ISSUE THREAD

**Issue by Mark-J-Lawrence** [ansible-collections/ibm_zos_core#540]
### Bug description

Playbook contains the following task:

```
- name: Submit job to regenerate RACF commands
      ibm.ibm_zos_core.zos_job_submit:
        src: ../racf.jcl
        location: LOCAL
        wait: true
        wait_time_s: 300
        return_output: true
```

When I run locally, it logs in as my own user (using a password) on the z/OS machine, and it works perfectly.

However, when I then run this as part of a jenkins job (using an SSH key to a user called `JENKINS`) it fails with an exception. I have tried changing the `location` to `USS`, but it fails with the same error.

The exception thrown is not particularly helpful because it appears another exception is thrown which doesn't inherit from `BaseException`, which causes a distraction.

To diagnose the problem myself a bit further, I manually changed `/python/v3r10/pyz/lib/python3.10/site-packages/zoautil_py/jobs.py` to stream `response.stdout_response` to a local text file (`response.stderr_response` was empty, the line that raises a ZOAUException does not handle this). The file contained:

```
submit 'JENKINS.P3622224.T0720544.C0000000'
RECORD LENGTH OF DATA SET JENKINS.P3622224.T0720544.C0000000 NOT 80
```

A quick google of that response sent me here: https://www.ibm.com/docs/en/zos/2.1.0?topic=ikj-ikj56261i

So it appears the temporary data set that is being created to store the JCL isn't being created with the correct record size....but only for the JENKINS user, for some reason?

If I change `location` to `DATA_SET` and point it to the JCL stored on the system, the task hangs.

### Playbook verbosity output

[ansible_log.docx](https://github.com/ansible-collections/ibm_zos_core/files/9829128/ansible_log.docx)

### Contents of `ansible.cfg`

```
[defaults]
inventory = ./inventory
library = ./library
action_plugins=./plugins/action
host_key_checking = False
pipelining = False
```

### Contents of the inventory

```
all:
  hosts:
    winmvsb0:
      ansible_host: winmvsb0.hursley.ibm.com
      ansible_user: jenkins
      ansible_connection: ssh 
      host_key_checking: false
```

### Contents of `group_vars` or `host_vars`

`host_vars`:

```
PYZ: "/python/v3r10/pyz"
ZOAU: "/usr/lpp/IBM/zoautil"
ansible_python_interpreter: "{{ PYZ }}/bin/python"

# The location of ROCKET z Open Source TOOLS
TOOLS_ROOT: "/rocket"
TOOLS_PATH: "{{ TOOLS_ROOT }}/bin"

GIT_SHELL: "{{ TOOLS_ROOT }}/bin/bash"
GIT_EXEC_PATH: "{{ TOOLS_ROOT }}/libexec/git-core"
GIT_TEMPLATE_DIR: "{{ TOOLS_ROOT }}/share/git-core/templates"
```

`group_vars`:

```
PYZ: "/python/v3r10/pyz"
ZOAU: "/usr/lpp/IBM/zoautil"

environment_vars:
  _BPXK_AUTOCVT: "ON"
  PYTHON: "{{ PYZ }}/bin"
  ZOAU_ROOT: "{{ ZOAU }}"
  PATH: "{{ ZOAU }}/bin:{{ PYZ }}/bin:/bin:/var/bin:/usr/sbin"
  LIBPATH: "{{ ZOAU }}/lib:{{ PYZ }}/lib:/lib:/usr/lib:."
  MANPATH: "{{ ZOAU }}/docs/%L"
  _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
  _TAG_REDIR_ERR: "txt"
  _TAG_REDIR_IN: "txt"
  _TAG_REDIR_OUT: "txt"
  LANG: "C"
```

### Ansible version

```
ansible-core==2.11.11
```

### IBM z/OS Ansible core Version

v1.4.0-beta.1

### IBM ZOAU version

v1.1.1

### z/OS version

z/OS 02.04.00

### Ansible module

zos_job_submit

**Comment by ddimatos** [comment]
@Mark-J-Lawrence thank for opening the issue, I will look into this to evaluate next steps and share that information here.

**Comment by ddimatos** [comment]
Hi @Mark-J-Lawrence - i did a bit of looking into this, I am hoping you can respond to some of my questions to help me better understand, thus I have some comments and questions. Also, thank you for all the extra debugging you did.
1) ZOAU_ROOT: "{{ ZOAU }}" has not been used since ZOAU 1.0.3, yet I don't think i matters as I just scanned the code and found no usage of it except in build scripts, just wanted to point out now its `ZOAU_HOME`
2) You stated you are on 1.4.0-beta.1 Ansible core but I am not able to align the error messages with the code, is it possible the ansible core collection installed on your pipeline is not 1.4.0.-beta.1 ? You can check with[ these instructions](https://ibm.github.io/z_ansible_collections_doc/faqs/faqs.html?highlight=grep#how-can-i-know-which-version-of-a-collection-i-am-using) and verify
3) The error where `line 647, in submit_uss_jcl calls ..... zoautil_py/jobs.py", line 58` is saying we are using ZOAU to submit jobs, we have not used ZOAU to submit JCL since our Ansible core 1.0, until recent in our code for Ansible core 1.5.0-beta.1 we plan to migrate back to ZOAU submit code. Which makes me wonder what version of the collectiosn your Jenkins node is running. 
4) Regarding "RECORD LENGTH OF DATA SET JENKINS.P3622224.T0720544.C0000000 NOT 80" , the default ZOAU data sets have an LRCL of 80, which covers any JCL permitted, so I do not know how the LRCL could be an issue, see next comment.
5) Although we don't use ZOAU for job submission till 1.5.0 once again, the error in bullet 4 could be a result of the user JENINS not having permission to create a data set with HLQ Jenkins, 1.4.0-beta.2 we don't use a data set for JCL, we submit the JCL as a file, so unsure how this could even be the issue for 1.4.0-beta.2 e.g, `jobId = submit_uss_jcl(temp_file_2.name, module)`
6) Just so you know, Python 3.10 is not tested/supported till Ansible core 1.5.0, if you continue to use Python 3.10 be sure to disable pipelineing in your ansible.cfg
7) Just so you know, the only ZOAU that supports Python 3.10 is ZOAU 1.2.0.1 or later, if you use any other combination you will get a `0X4` - `0X8` unicode error, I built ZOAU 1.2.01 specifically to support python 3.10 so i can confirm this experience. 
8) If for some reason the HLQ is an issue for user JENKINS, you can override that with an environment variable like so:
```
environment_vars:
  _BPXK_AUTOCVT: "ON"
  .....
  ......
  TMPHLQ: "FOO"
```
9) The ZOAU exception that is not reporting anything, I requested that to be fixed, it should be fixed in ZOAU 1.2.1. 


Could you:
- confirm your Ansible core version (bullet 2) on the Jenkins node
- consider pointing to another supported version of Python (eg 3.8.x or 3.9.x)
- Use ISPF or ZOAU `dls` to list the Data set Attributes for `JENKINS.P3622224.T0720544.C0000000`
- Provide me the output using unedited source `ANSIBLE_DEBUG=1 ansible-playbook -i inventory your-playbook.yml -vvvv`

Thank you.

**Comment by ddimatos** [comment]
FYI @Mark-J-Lawrence 

Over 60 days old, if you need to reopen, please do so as a new git issue ensuring you have the latest release of IBM z/OS Core and the required dependencies because at this time, z/OS Ansible Core version 1.4.0 has GA'd and soon 1.5.0-beta.1 will.
